src/Symfony/Bundle/FrameworkBundle/Test/KernelTestCase.php

```diff
@@ -39,6 +39,12 @@ protected function tearDown(): void
         static::$booted = false;
     }
 
+    public static function tearDownAfterClass(): void
+    {
+        static::ensureKernelShutdown();
+        parent::tearDownAfterClass();
```

Comment:
This should do the same as tearDown to avoid leaving the class in a dirty state.

```suggestion
        parent::tearDownAfterClass();
        static::$class = null;
        static::$kernel = null;
        static::$booted = false;
```

Without this, at least static::$kernel will remain set and that is what caused the issue for me on the other PR.

Reply:
+ no needs to call parent::tearDownAfterClass, it's a no-op

Reply:
Added @Seldaek suggestion and removed the `parent::tearDownAfterClass()` no-op call.

I thought about grouping 

```
static::$class = null;
static::$kernel = null;
static::$booted = false;
```

in a `private static function clearTestState`, wdyt?